src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -0,0 +1,309 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\CustomUrl\Manager;
+
+use Ferrandini\Urlizer;
+use PHPCR\ItemExistsException;
+use PHPCR\Util\PathHelper;
+use Sulu\Component\CustomUrl\Document\CustomUrlDocument;
+use Sulu\Component\CustomUrl\Document\RouteDocument;
+use Sulu\Component\CustomUrl\Repository\CustomUrlRepository;
+use Sulu\Component\DocumentManager\Behavior\Mapping\UuidBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\Exception\DocumentNotFoundException;
+use Sulu\Component\DocumentManager\MetadataFactoryInterface;
+use Sulu\Component\DocumentManager\PathBuilder;
+use Sulu\Component\HttpCache\HandlerInvalidatePathInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * Manages custom-url documents and their routes.
+ */
+class CustomUrlManager implements CustomUrlManagerInterface
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var CustomUrlRepository
+     */
+    private $customUrlRepository;
+
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    /**
+     * @var PathBuilder
+     */
+    private $pathBuilder;
+
+    /**
+     * @var HandlerInvalidatePathInterface
+     */
+    private $cacheHandler;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        CustomUrlRepository $customUrlRepository,
+        MetadataFactoryInterface $metadataFactory,
+        PathBuilder $pathBuilder,
+        HandlerInvalidatePathInterface $cacheHandler
+    ) {
+        $this->documentManager = $documentManager;
+        $this->customUrlRepository = $customUrlRepository;
+        $this->metadataFactory = $metadataFactory;
+        $this->pathBuilder = $pathBuilder;
+        $this->cacheHandler = $cacheHandler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, array $data, $locale = null)
+    {
+        $document = $this->documentManager->create('custom_url');
+        $this->bind($document, $data, $locale);
+
+        try {
+            $this->documentManager->persist(
+                $document,
+                $locale,
+                [
+                    'parent_path' => $this->getItemsPath($webspaceKey),
+                    'node_name' => Urlizer::urlize($document->getTitle()),
```

Comment:
it doesn't work here at all ... if i implement the "AutonameBehavior" it also implements the "TitleBehavior"which triggers the TitleSubscriber which uses (in a weird way) the StructureBehavior to determine the title. this doesnot work with the custom-url because it doesnt implement the StructureBehavior.